examples/utils/readcsv.py

```diff
@@ -0,0 +1,121 @@
+from pathlib import Path
+from typing import Any, Literal, Optional, Protocol, Sequence, TypeVar
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+
+DType = TypeVar(
+    "DType", type(np.float32), type(np.float64), Literal["float32"], Literal["float64"]
+)
+
+
+class ReadCsvFunc(Protocol):
+    """Protocol of CSV read functions with the necessary `kwargs`s for testing"""
+
+    def __call__(
+        self,
+        file_path: Path,
+        usecols: Optional[Sequence[int]] = None,
+        skip_header: int = 0,
+        dtype: DType = np.float32,
+        max_rows: Optional[int] = None,
+    ) -> Any:
+        ...
+
+
+class NpReadCsv:
```

Comment:
what is advantage of making these classes and not functions?

Reply:
See the inline comment where I defined the functions. I'm defining a protocol class. It makes type checking easier.